src/utils/printing/index.ts

```diff
@@ -65,20 +76,31 @@ export function makeDeviceChangeHandler(config: PrintConfig) {
       return
     }
 
+    const namedPrinterDeviceURI = await getPrinterDeviceURI(config.printerName)
+
+    if (namedPrinterDeviceURI === deviceURI) {
+      debug('printer "%s" is already configured, skipping', config.printerName)
+      return
+    }
+
     await configurePrinter({
       printerName: config.printerName,
       ppd: printer.ppd,
       deviceURI,
       setDefault: true,
     })
+
+    debug('re-triggering device add event after configuring printer')
+    currentListener.remove()
+    onDeviceChange.trigger(changeType, device)
+
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    currentListener = onDeviceChange.add(onDeviceChangeCallback)
```

Comment:
This approach is probably not needed since I added the short circuit when the printer is already configured. It’ll even be less noisy since this makes it fire for all the devices again.

Reply:
Fixed.